stream_alert/rule_processor/handler.py

```diff
@@ -106,16 +114,22 @@ def run(self, event):
             if not payload:
                 continue
 
-            self._process_alerts(payload)
+            self._run_batches(payload, workers)
+
+        LOGGER.debug('Number of running workers: %d', len(workers))
 
-        LOGGER.debug('Invalid record count: %d', self._failed_record_count)
+        # Wait for all of the workers to finish before continuing
+        for worker in workers:
+            worker.join()
+
+        LOGGER.debug('Invalid record count: %d', self._failed_record_count.value)
```

Comment:
This line is not new and is how we have it configured right now. @mime-frame is not a fan of conditional logging statements so if it's not a huge deal I'll leave it like this